invertedai/large/drive.py

```diff
@@ -0,0 +1,314 @@
+        Location name in IAI format.
+
+    agent_states:
+        Current states of all agents.
+        The state must include x: [float], y: [float] coordinate in meters
+        orientation: [float] in radians with 0 pointing along x and pi/2 pointing along y and
+        speed: [float] in m/s.
+
+    agent_attributes:
+        Static attributes of all agents.
+        List of agent attributes. Each agent requires, length: [float]
+        width: [float] and rear_axis_offset: [float] all in meters. agent_type: [str],
+        currently supports 'car' and 'pedestrian'.
+        waypoint: optional [Point], the target waypoint of the agent.
+
+    recurrent_states:
+        Recurrent states for all agents, obtained from the previous call to
+        :func:`drive` or :func:`initialize`.
+
+    traffic_lights_states:
+       If the location contains traffic lights within the supported area,
+       their current state should be provided here. Any traffic light for which no
+       state is provided will have a state generated by iai.
+
+    light_recurrent_states:
+       Light recurrent states for all agents, obtained from the previous call to
+        :func:`drive` or :func:`initialize`.
+       Specifies the state and time remaining for each light group in the map.
+       If manual control of individual traffic lights is desired, modify the relevant state(s) 
+       in traffic_lights_states, then pass in light_recurrent_states as usual.
+
+    get_birdview:
+        Whether to return an image visualizing the simulation state.
+        This is very slow and should only be used for debugging.
+
+    rendering_center:
+        Optional center coordinates for the rendered birdview.
+
+    rendering_fov:
+        Optional fov for the rendered birdview.
+
+    get_infractions:
+        Whether to check predicted agent states for infractions.
+        This introduces some overhead, but it should be relatively small.
+
+    random_seed:
+        Controls the stochastic aspects of agent behavior for reproducibility.
+
+    api_model_version:
+        Optionally specify the version of the model. If None is passed which is by default, 
+        the best model will be used.
+    
+    capacity:
+        The number of agents allowed in a region before it must subdivide. Currently this 
+        value represents the capacity of a quadtree leaf node that will subdivide if the 
+        number of vehicles in the region passes this threshold.
+
+    is_async:
+        A flag to control whether to use asynchronous DRIVE calls.
+
+    See Also
+    --------
+    :func:`drive`
+    """
+    
+    num_agents = len(agent_states)
+    agent_x = [None]*num_agents
+    agent_y = [None]*num_agents
+    for i, agent in enumerate(agent_states):
+        agent_x[i] = agent.center.x
+        agent_y[i] = agent.center.y
+    max_x, min_x, max_y, min_y = max(agent_x), min(agent_x), max(agent_y), min(agent_y)
+    region_size = ceil(max(max_x,max_y) - min(min_x,min_y))
```

Comment:
The reason this is rounded up to the nearest integer is to avoid issues with floating points as the region size is divided over and over. In testing, this would create a small, but non-zero gap, between leaf nodes in the quad tree and this was the best option for solving that issue. 

Otherwise the change is good and implemented.